vendor-extra/LiberoPageBundle/src/DependencyInjection/LiberoPageConfiguration.php

```diff
@@ -45,12 +45,27 @@ private function getPagesDefinition() : ArrayNodeDefinition
         $pagesNode
             ->arrayPrototype()
                 ->children()
-                    ->scalarNode('path')
+                    ->enumNode('type')
+                        ->values(['content', 'homepage'])
                         ->isRequired()
                     ->end()
-                    ->scalarNode('service')
+                    ->scalarNode('path')
                         ->isRequired()
                     ->end()
+                    ->scalarNode('content_service')
+                    ->end()
+                ->end()
+                ->validate()
+                    ->ifTrue(function (array $values) : bool {
+                        return 'content' === $values['type'] && !isset($values['content_service']);
+                    })
+                    ->thenInvalid('Content pages require a content_service')
+                ->end()
+                ->validate()
+                    ->ifTrue(function (array $values) : bool {
+                        return 'content' !== $values['type'] && isset($values['content_service']);
+                    })
+                    ->thenInvalid('Non-content pages cannot have a content_service')
```

Comment:
This is a bit messy. Don't know if there's a better way to have a discriminator value (`type`), but maybe the different types should have their own configuration instead.

Reply:
Maybe better to group them by type, so something like:

```yaml
libero_page:
    pages:
        homepage:
            name: 'homepage'
            path: '/'
        content:
            - name: blog_article
              path: '/blog/{id}'
              content_service: 'blog-articles'
            - name: scholarly_article:
              path: '/articles/{id}'
              content_service: 'scholarly-articles'
```

So each group can either be a single entry, or a list of entries.